src/viperleed/utilities/poscar/modify_vacuum.py

```diff
@@ -55,42 +72,35 @@ def modify_vacuum(slab, vacuum_gap_size, absolute=False):
 
     slab_thickness = processed_slab.thickness
     current_gap_size = processed_slab.vacuum_gap
+    vacuum_gap_size = vacuum_gap_info.size
 
-    vacuum_gap_size += current_gap_size if not absolute else 0
+    vacuum_gap_size += current_gap_size if not vacuum_gap_info.absolute else 0
 
-    # check there is enough space on top of the slab to reduce the vacuum gap
-    if vacuum_gap_size < current_gap_size:
-        top_atom_positions = max(at.cartpos[2] for at in slab)
-        gap_on_top = slab.c_vector[2] - top_atom_positions
-        if gap_on_top < (current_gap_size - vacuum_gap_size):
-            raise RuntimeError('The resulting vacuum gap size would be larger '
-                               'than the distance between the topmost atom and '
-                               'the top of the cell.')
+    if vacuum_gap_size < 0:
+        raise NotEnoughVacuumError(
+            "The resulting vacuum gap size would be negative.", None)
 
     logger.debug(f'Current vacuum gap size:\t{current_gap_size:9.3f}')
     logger.debug(f'New vacuum gap size:\t\t{vacuum_gap_size:9.3f}')
 
-    if vacuum_gap_size < 0:
-        raise RuntimeError('The resulting vacuum gap size would be negative.')
-
     new_c_vector = (
         processed_slab.c_vector
         / np.linalg.norm(processed_slab.c_vector)
         * (vacuum_gap_size + slab_thickness)
     )
-    print(f'New c vector: {new_c_vector}')
-    print(f'Old c vector: {processed_slab.c_vector}')
-
     processed_slab.c_vector[:] = new_c_vector
     processed_slab.collapse_cartesian_coordinates()
```

Comment:
Now that I think about it, I feel like you probably need an `update_origin=True` in here, since atoms do move along _z_ as a result of the resizing of the **c** vector.